packages/game-server/src/shared/extensions/interactive.ts

```diff
@@ -0,0 +1,51 @@
+import { GenericEntity } from "../entities";
+import { Player } from "../entities/player";
+import { Extension, ExtensionNames, ExtensionSerialized } from "./types";
+
+interface InteractiveOptions {
+  eventName?: string;
+}
+
+export default class Interactive implements Extension {
+  public static readonly Name = ExtensionNames.interactive;
+
+  private self: GenericEntity;
+  private eventName: string | null = null;
+
+  public constructor(self: GenericEntity) {
+    this.self = self;
+  }
+
+  public init(options: InteractiveOptions): this {
+    if (options.eventName !== undefined) {
+      this.eventName = options.eventName;
+    }
+    return this;
+  }
+
+  public interact(player: Player): void {
+    if (player.isInventoryFull()) {
```

Comment:
this feels wrong, what if a button entity is "interactive"? why would it need to check the player's inventory.